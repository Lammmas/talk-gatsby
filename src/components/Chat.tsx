import React from 'react';
import Talk from 'talkjs';

const chatBoxStyle = {
    height: '500px',
};

export default class Chat extends React.Component {
    session: Talk.Session | null = null;
    chatboxEl: any = React.createRef();

    async componentDidMount() {
        await Talk.ready;

        const currentUser = new Talk.User({
            id: '1',
            name: 'Henry Mill',
            email: 'henrymill@example.com',
            photoUrl: 'henry.jpeg',
            welcomeMessage: 'Hello!',
            role: 'default',
        });
        const otherUser = new Talk.User({
            id: '2',
            name: 'Jessica Wells',
            email: 'jessicawells@example.com',
            photoUrl: 'jessica.jpeg',
            welcomeMessage: 'Hello!',
            role: 'default',
        });

        this.session = new Talk.Session({
            appId: 'tXAzIgN4', // TODO: Replace with env var
            me: currentUser,
        });

        const conversationId = Talk.oneOnOneId(currentUser, otherUser);
        const conversation = this.session.getOrCreateConversation(conversationId);
        conversation.setParticipant(currentUser);
        conversation.setParticipant(otherUser);

        const chatbox = this.session.createChatbox();
        chatbox.select(conversation);
        chatbox.mount(this.chatboxEl.current);
    }

    render() {
        return <div style={chatBoxStyle} ref={this.chatboxEl} />;
    }

    componentWillUnmount() {
        if (this.session) this.session.destroy();
    }
}
