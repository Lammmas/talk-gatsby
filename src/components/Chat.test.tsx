import React from 'react';
import renderer from 'react-test-renderer';

// Note: This is so in debth for future testing of use cases
jest.mock('talkjs', () => {
    return {
        __esModule: true,
        default: {
            ready: Promise.resolve(true),
            User: jest.fn(),
            Session: jest.fn(() => ({
                getOrCreateConversation: jest.fn(() => ({
                    setParticipant: jest.fn(),
                })),
                createChatbox: jest.fn(() => ({
                    select: jest.fn(),
                    mount: jest.fn(),
                })),
            })),
            oneOnOneId: jest.fn(),
        },
    };
});

import Chat from './Chat';

describe('Chat', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Chat />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})