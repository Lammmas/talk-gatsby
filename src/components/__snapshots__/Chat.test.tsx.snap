// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat renders correctly 1`] = `
<div
  style={
    {
      "height": "500px",
    }
  }
/>
`;
